src/main/java/com/marklogic/contentpump/RDFReader.java

```diff
@@ -1044,8 +1044,12 @@ private String formatMessage(String message, long line, long col) {
 
         @Override
         public void warning(String message, long line, long col) {
-            //For Bug 24519, to improve readability of logs, I have chose to log the warnings for IRI only in debug mode. 
-            if (message.contains("Bad IRI:") || message.contains("Illegal character in IRI")) {
+            //For Bug 24519, to improve readability of logs, I have chose to log the warnings for IRI only in debug mode.
+            List<String> debugwarninglist = new ArrayList<>();
```

Comment:
One suggestion is to use HashSet other than List, due to the search performance.

Reply:
Excellent point, will do it.